js/samples/08.datasource.azureAISearch/src/AzureAISearchDataSource.ts

```diff
@@ -0,0 +1,233 @@
+    indexName: string;
+
+    /**
+     * Azure OpenAI API key. AzureO
+     */
+    azureOpenAIApiKey: string;
+
+    /**
+     * Azure OpenAI endpoint. This is used to generate embeddings for the user's input.
+     */
+    azureOpenAIEndpoint: string;
+
+    /**
+     * Azure OpenAI Embedding deployment. This is used to generate embeddings for the user's input.
+     */
+    azureOpenAIEmbeddingDeployment: string;
+
+    /**
+     * Azure AI Search API key.
+     */
+    azureAISearchApiKey: string;
+
+    /**
+     * Azure AI Search endpoint.
+     */
+    azureAISearchEndpoint: string;
+}
+
+/**
+ * A data source that uses a Azure AI Search index to inject text snippets into a prompt.
+ */
+export class AzureAISearchDataSource implements DataSource {
+    /**
+     * Name of the data source and local index.
+     */
+    public readonly name: string;
+
+    /**
+     * Options for creating the data source.
+     */
+    private readonly options: AzureAISearchDataSourceOptions;
+
+    /**
+     * Azure AI Search client.
+     */
+    private readonly searchClient: SearchClient<Restaurant>;
+
+    /**
+     * Creates a new `AzureAISearchDataSource` instance.
+     * @param {AzureAISearchDataSourceOptions} options Options for creating the data source.
+     */
+    public constructor(options: AzureAISearchDataSourceOptions) {
+        this.name = options.name;
+        this.options = options;
+        this.searchClient = new SearchClient<Restaurant>(
+            options.azureAISearchEndpoint,
+            options.indexName,
+            new AzureKeyCredential(options.azureAISearchApiKey),
+            {}
+        );
+    }
+
+    /**
+     * Renders the data source as a string of text.
+     * @remarks
+     * The returned output should be a string of text that will be injected into the prompt at render time.
+     * @param {TurnContext} context Turn context for the current turn of conversation with the user.
+     * @param _context
+     * @param {Memory} memory An interface for accessing state values.
+     * @param {Tokenizer} tokenizer Tokenizer to use when rendering the data source.
+     * @param {number} maxTokens Maximum number of tokens allowed to be rendered.
+     * @returns {Promise<RenderedPromptSection<string>>} A promise that resolves to the rendered data source.
+     */
+    public async renderData(
+        _context: TurnContext,
+        memory: Memory,
+        tokenizer: Tokenizer,
+        maxTokens: number
+    ): Promise<RenderedPromptSection<string>> {
+        const query: string = memory.getValue('temp.input');
+
+        // If the user input is empty, don't add any text to the prompt.
+        if (!query) {
+            return { output: '', length: 0, tooLong: false };
+        }
+
+        const selectedFields = [
+            'restaurantId',
+            'restaurantName',
+            'description',
+            'category',
+            'tags',
+            'rating',
+            'location',
+            'link',
```

Comment:
`link` is not defined in the index, so it fails actually with this. Suggesting to remove this field from the selection ;)

BTW thanks for this big PR, I was about to start working on that, it saves me a lot of time,